firefox-ios/Client/Frontend/Home/LegacyHomepageViewController.swift

```diff
@@ -289,13 +289,20 @@ class LegacyHomepageViewController:
     }
 
     func createLayout() -> UICollectionViewLayout {
-        // swiftlint:disable line_length
-        let layout = UICollectionViewCompositionalLayout { [weak self] (sectionIndex: Int, layoutEnvironment: NSCollectionLayoutEnvironment) -> NSCollectionLayoutSection? in
-        // swiftlint:enable line_length
-            guard let self = self,
+        let zeroLayoutSize = NSCollectionLayoutSize(widthDimension: .absolute(0.0), heightDimension: .absolute(0.0))
+        let emptyGroup = NSCollectionLayoutGroup.horizontal(layoutSize: zeroLayoutSize,
+                                                            subitems: [NSCollectionLayoutItem(layoutSize: zeroLayoutSize)])
+        let emptyDefaultLayoutSection = NSCollectionLayoutSection(group: emptyGroup)
```

Comment:
Can you add this block to a helper function that only gets called where you return emptyDefaultLayoutSection. That way it cleans up the code a little and doesn't instantiate these objects until they are actually needed.

Reply:
Yes sure, i'm going to do it now